platform/engineering/design-docs/2020-12-10-auto-deploy-cms-build.md

```diff
@@ -0,0 +1,195 @@
+
+Both deployments will be essential to achieve the goal:
+
+1. Webpack auto-deploy once per day (This already exists)
+   - [Content build Jenkinsfile](https://github.com/department-of-veterans-affairs/vets-website/blob/master/Jenkinsfile#L63)
+2. Content-only auto-deploy every 5 minutes (Needs to be implemented, interval time may vary)
+   - [Content-only build Jenkinsfile](https://github.com/department-of-veterans-affairs/vets-website/blob/2b8c478ebe48882ca098a7c8f4cf684c69fb3114/Jenkinsfile.content#L72)
+
+Both Jenkins files are called from `common.groovy`
+
+#### Pain points
+
+- Engineers encounter confusing build errors that are due to the mismatch in outdated locally cached content and updated Liquid templates
+- Failure using the latest CMS content
+- CMS triggers content build manually
+- Content writers want to quickly draft, publish, and deploy content. However, `vets-website` is only deployed to production once per day. To allow content writers to move faster, there is a partial deploy, which only includes static page changes (vagov-content and Drupal).
+- No accessibility checking
+- No broken links checking
+
+### High Level Design
+
+Once the CMS build finishes successfully and all the transformers work is complete, there will be a transition from the GraphQL build to the CMS build in Metalsmith.
+
+Metalsmith will pass the CMS flag so the CMS content build will be called.
+
+There are three things that will be to happen:
+
+1. A full deploy
+2. A recurrent partial deploy (every 5 mins ideally)
+3. A script to verify the CMS content
+
+## Specifics
+
+### Detailed Design
+
+#### A full deploy
+
+To deploy the CMS content build inside the current auto-deploy for `vets-website` the `--use-cms-export` flag will need to be passed to the `build` function inside `common.groovy`
+
+#### A recurrent partial deploy (every 5 mins ideally)
+
+In addition to the full deploy, a partial deploy needs to be fired up to create a content-only build and update the AWS bucket with the newest content.
+
+Currently, the CMS team is doing this process manually. They have been deploying content-only as needed.
+
+CMS team has recommended an auto-deploy with 5 mins intervals while they provide support.
+
+The idea here is to create a Jenkins/CircleCI job to run a cron job that will trigger a partial deploy, initially, every hour from 8am-8pm EST. Depending on the success rate of the build, the intervals time will be reduced.
+
+#### A script to verify the CMS content
+
+Deploying the partial build every X minutes is not very proficient, therefore, there needs to be a script that verifies that the CMS content is not the same.
+
+#### When to deploy the content-only build
+
+There needs to be a script that will:
+
+- Download the CMS content (`.tar`) file and will compare it with the previous one
```

Comment:
Will this be from https://prod.cms.va.gov/cms-export/content? Maybe we should add where we'll be downloading the CMS content for context. Also, will we be getting the previous export from S3?